src/app/modules/main/chat_section/view.nim

```diff
@@ -148,7 +148,9 @@ QtObject:
 
   proc activeItemSet*(self: View, item: Item) =
     self.activeItem.setActiveItemData(item)
-    self.activeItemChanged()
+    var time = cpuTime()
+    self.activeItemChanged() # UI block happens after emitting this signal
```

Comment:
Attaching video of behavior with `activeItemChanged()` commented out. Chat loading is instant and by checking which slot is connected to `activeItemChanged`, namely:
https://github.com/status-im/status-desktop/pull/12590/commits/aa0d650585d40a4bc89f33168268bd3b58269330#diff-0c5cf8e090e842349eba2a02db4d37e596e51e0c1788151d361984a1084c0c09R102-R103

It might be related to `CreateChatView` new persistent module introduced earlier, since we seem to re-instantiate a new store `root.rootStore.openCreateChat` for each chat view switch?

https://github.com/status-im/status-desktop/assets/544446/75eb499b-6eb1-4956-8cd6-1bde0edce43b

Reply:
If you follow the trail a bit further, you will notice that `openCreateChat` is responsible for the [visibility](https://github.com/status-im/status-desktop/blob/7fe6e10b814380e3a12e48ddccf1b6db9b6d6229/ui/app/AppLayouts/Chat/views/ChatColumnView.qml#L200) of `ChatContentView`. I would guess the problem lies somewhere there. The visibility change may trigger many things; therefore, you need to peel back the layers further to uncover it :crossed_fingers: